### ISSUE TITLE
PatternSyntaxException: Unexpected internal error near index 1 

### ISSUE THREAD

**Issue by pihme** [camunda-cloud/zeebe#7347]
**Describe the bug**
Observed in production error logs:
https://console.cloud.google.com/errors/CNf71d_o46X10gE?service=zeebe&time=P7D&refresh=off&project=camunda-cloud-240911

This lead to a blacklisting of the process instance.

**Log/Stacktrace**

<details><summary>Full Stacktrace</summary>
 <p>

```
java.util.regex.PatternSyntaxException: Unexpected internal error near index 1
\
	at java.util.regex.Pattern.error(Unknown Source) ~[?:?]
	at java.util.regex.Pattern.compile(Unknown Source) ~[?:?]
	at java.util.regex.Pattern.<init>(Unknown Source) ~[?:?]
	at java.util.regex.Pattern.compile(Unknown Source) ~[?:?]
	at org.camunda.feel.impl.builtin.StringBuiltinFunctions$$anonfun$splitFunction$1.applyOrElse(StringBuiltinFunctions.scala:179) ~[feel-engine-1.12.4.jar:1.12.4]
	at org.camunda.feel.impl.builtin.StringBuiltinFunctions$$anonfun$splitFunction$1.applyOrElse(StringBuiltinFunctions.scala:177) ~[feel-engine-1.12.4.jar:1.12.4]
	at scala.PartialFunction$OrElse.apply(PartialFunction.scala:241) ~[scala-library-2.13.4.jar:?]
	at org.camunda.feel.impl.interpreter.FeelInterpreter.invokeFunction(FeelInterpreter.scala:775) ~[feel-engine-1.12.4.jar:1.12.4]
	at org.camunda.feel.impl.interpreter.FeelInterpreter.$anonfun$eval$40(FeelInterpreter.scala:189) ~[feel-engine-1.12.4.jar:1.12.4]
	at org.camunda.feel.impl.interpreter.FeelInterpreter.withFunction(FeelInterpreter.scala:742) ~[feel-engine-1.12.4.jar:1.12.4]
	at org.camunda.feel.impl.interpreter.FeelInterpreter.eval(FeelInterpreter.scala:189) ~[feel-engine-1.12.4.jar:1.12.4]
	at org.camunda.feel.impl.interpreter.FeelInterpreter.$anonfun$withCartesianProduct$1(FeelInterpreter.scala:823) ~[feel-engine-1.12.4.jar:1.12.4]
	at scala.collection.immutable.List.map(List.scala:246) ~[scala-library-2.13.4.jar:?]
	at org.camunda.feel.impl.interpreter.FeelInterpreter.withCartesianProduct(FeelInterpreter.scala:823) ~[feel-engine-1.12.4.jar:1.12.4]
	at org.camunda.feel.impl.interpreter.FeelInterpreter.eval(FeelInterpreter.scala:145) ~[feel-engine-1.12.4.jar:1.12.4]
	at org.camunda.feel.impl.interpreter.FeelInterpreter.$anonfun$eval$2(FeelInterpreter.scala:57) ~[feel-engine-1.12.4.jar:1.12.4]
	at org.camunda.feel.impl.interpreter.FeelInterpreter.$anonfun$foldEither$2(FeelInterpreter.scala:234) ~[feel-engine-1.12.4.jar:1.12.4]
	at scala.util.Either$RightProjection.flatMap(Either.scala:757) ~[scala-library-2.13.4.jar:?]
	at org.camunda.feel.impl.interpreter.FeelInterpreter.$anonfun$foldEither$1(FeelInterpreter.scala:234) ~[feel-engine-1.12.4.jar:1.12.4]
	at scala.collection.LinearSeqOps.foldLeft(LinearSeq.scala:169) ~[scala-library-2.13.4.jar:?]
	at scala.collection.LinearSeqOps.foldLeft$(LinearSeq.scala:165) ~[scala-library-2.13.4.jar:?]
	at scala.collection.immutable.List.foldLeft(List.scala:79) ~[scala-library-2.13.4.jar:?]
	at org.camunda.feel.impl.interpreter.FeelInterpreter.foldEither(FeelInterpreter.scala:233) ~[feel-engine-1.12.4.jar:1.12.4]
	at org.camunda.feel.impl.interpreter.FeelInterpreter.eval(FeelInterpreter.scala:59) ~[feel-engine-1.12.4.jar:1.12.4]
	at org.camunda.feel.FeelEngine.eval(FeelEngine.scala:200) ~[feel-engine-1.12.4.jar:1.12.4]
	at org.camunda.feel.FeelEngine.$anonfun$eval$2(FeelEngine.scala:196) ~[feel-engine-1.12.4.jar:1.12.4]
	at scala.util.Either.flatMap(Either.scala:352) ~[scala-library-2.13.4.jar:?]
	at org.camunda.feel.FeelEngine.eval(FeelEngine.scala:196) ~[feel-engine-1.12.4.jar:1.12.4]
```

</p>
</details>

**Environment:**
Camunda Cloud, 0.26.4

**Comment by npepinpe** [comment]
Let's check if this is still an issue in the latest versions.

**Comment by saig0** [comment]
I created an issue in the FEEL engine to handle the failure: https://github.com/camunda/feel-scala/issues/289

However, it is not clear which expression caused the problem :thinking:

**Comment by Zelldon** [comment]
@saig0 we could find out with zdb

**Comment by saig0** [comment]
The issue camunda/feel-scala#289 is fixed and will be included in the next patch release `1.13.3` :rocket:

**Comment by saig0** [comment]
The issue should be fixed by #7682 (dump `feel-engine` to `1.13.3`). The creation of the pattern is wrapped in a `try-catch` block and will return a failure instead.
